index.js

```diff
@@ -11,11 +12,44 @@ server.get('/payments', (req, res) => {
     res.status(200).send(payments);
 });
 
+server.get('/payments/:id', (req, res) => {
+    const payment = payments.find( payment => payment.id === req.params.id);
+    if(payment){
+        res.status(200).send(payment);
+        return;
+    }
+    res.status(404).send({message: 'Wrong id'});
+});
+
+server.post('/payments', (req, res) => {
+    const payment = {...req.body};
+    Object.assign(payment, {
+        id: uuidv4(),
+        status: 'Pending'
+    });
+    payments.push(payment);
+    res.status(200).send(payment);
+});
 
+server.put('/payments/:id', (req, res) => {
+    const payment = payments.find( payment => payment.id === req.params.id);
+    if(!payment || !req.body.description){
+        res.status(404).send('Wrong description or server can not find payment with this ID');
+        return;
+    }
+    payment.description = req.body.description;
+    res.status(200).send(payment);
+});
+
+server.delete('/payments/:id', (req, res) => {
+    const payment = payments.find( payment => payment.id === req.params.id);
+    const paymentIdx = payments.indexOf(payment);
```

Comment:
2 lines above could be combined in one-liner if we use [Array.findIndex()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex)